### PR TITLE
Add separate test suite for openshift only examples

### LINKED ISSUE #996: Create a suite of e2e-example-tests for OpenShift only tests
Start with the TestSimpleProdDeployEsExample that was removed as part of https://github.com/jaegertracing/jaeger-operator/pull/995

Putting openshift only tests in their own suite can help avoid resource problems on github that we ran into as part of #974

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

This fixes #996

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @pavolloffay @rubenvp8510 Please review

File: test/e2e/examples_openshift_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+type ExamplesTestSuiteOCP struct {
+	suite.Suite
+}
+
+func (suite *ExamplesTestSuiteOCP) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
+		t.Skip("This wuite should only be run on OpenShift")
```
Comment (rubenvp8510) [inline test/e2e/examples_openshift_test.go]: typo, wuite => suite.

File: Makefile [hunk 2]
```diff
@@ -83,7 +83,7 @@ unit-tests:
 	@go test $(VERBOSE) $(UNIT_TEST_PACKAGES) -cover -coverprofile=cover.out
 
 .PHONY: e2e-tests
-e2e-tests: prepare-e2e-tests e2e-tests-smoke e2e-tests-cassandra e2e-tests-es e2e-tests-self-provisioned-es e2e-tests-streaming e2e-tests-examples1 e2e-tests-examples2
+e2e-tests: prepare-e2e-tests e2e-tests-smoke e2e-tests-cassandra e2e-tests-es e2e-tests-self-provisioned-es e2e-tests-streaming e2e-tests-examples1 e2e-tests-examples2  e2e-tests-examples-openshift
```
Comment (objectiser) [inline Makefile]: nit: two spaces before `e2e-tests-examples-openshift`

**Comment by kevinearls** [comment]
@objectiser CI has passed, can you merge?

**Pull request merged**